File: LemonTracker/Models/NoteData.cs
using LemonTracker.Enums;

namespace LemonTracker.Models;

public class NoteData
{
    public Note Note { get; set; }
    public int Octave { get; set; }
    
    public override string ToString()
    {
        var noteString = Note switch
        {
            Note.C => "C-",
            Note.Csharp => "C#",
            Note.D => "D-",
            Note.Dsharp => "D#",
            Note.E => "E-",
            Note.F => "F-",
            Note.Fsharp => "F#",
            Note.G => "G-",
            Note.Gsharp => "G#",
            Note.A => "A-",
            Note.Asharp => "A#",
            Note.B => "B-",
            Note.Off => "R-",
            _ => "--"
        };
        
        return $"{noteString}{(Note == Note.None ? "-" : Octave)}";
    }
}

File: src/models/note-data.js
export default class NoteData {
  constructor(note, octave) {
    this.note = note;
    this.octave = octave;
  }

  toString() {
    return `${this.note}${
      this.note === Note.None || this.note === Note.Off ? "-" : this.octave
    }`;
  }
}

export const Note = {
  None: "--",
  C: "C-",
  Csharp: "C#",
  D: "D-",
  Dsharp: "D#",
  E: "E-",
  F: "F-",
  Fsharp: "F#",
  G: "G-",
  Gsharp: "G#",
  A: "A-",
  Asharp: "A#",
  B: "B-",
  Off: "R-",
};
